controllers/src/main/java/com/redhat/ipaas/controllers/integration/IntegrationController.java

```diff
@@ -74,13 +72,17 @@ public void start() {
             if (event!=null && "change-event".equals(event)) {
                 try {
                     ChangeEvent changeEvent = Json.mapper().readValue(data, ChangeEvent.class);
-                    if ((changeEvent != null
-                        || changeEvent.getKind().isPresent() && changeEvent.getId().isPresent())
-                        && (Kind.from(changeEvent.getKind().get()) == Kind.Integration)) {
-                        enqueueCheckIntegration(changeEvent.getId().get());
+                    if (changeEvent != null) {
+                        changeEvent.getId().ifPresent(
+                            id -> changeEvent.getKind().ifPresent(
+                                kind -> {
+                                    if (Kind.from(kind) == Kind.Integration) {
+                                        enqueueCheckIntegration(id);
```

Comment:
I think the better way to do this is to have a reconciliation loop that reconciles desired vs actual state. This of course introduces a potential delay between requesting stuff and stuff actually happening, but I think that is acceptable as long as the loop interval is small enough. It reduces the requirement to know the state of running checks, plus allows failing over to a redundant controller without having to share state between processes.